lexibank_bodtkhobwa.py

```diff
@@ -126,19 +118,29 @@ def cmd_install(self, **kw):
                     }
 
             # add data to cldf
+            ds['FormTable', 'Segments'].separator = ' + '
+            ds['FormTable', 'Segments'].datatype = Datatype.fromvalue({
+                "base": "string",
+                "format": "([\\S]+)( [\\S]+)*"
+            })
             for idx in tqdm(data, desc='cldf the data'):
-                segments = [mapper.get(x, x) for x in data[idx, 'tokens']]
-                segments = ' '.join(segments).split()
+                segments = ' '.join([mapper.get(x, x) for x in data[idx, 'tokens']])
+                morphemes = segments.split(' + ')
                 concept = concepts.get(data[idx, 'concept'], '')
-                ds.add_lexemes(
+                for lex in ds.add_lexemes(
                     Language_ID=data[idx, 'doculect'].lower(),
                     Parameter_ID=concept,
                     Form=data[idx, 'form'],
                     Value=data[idx, 'value'],
-                    Segments=segments,
+                    Segments=morphemes,
                     Source=['Bodt2019'],
                     #PhoneticValue=vals['phonetic']
-                    )
-
-
-
+                ):
+                    for morpheme_index, cogid in enumerate(data[idx, 'crossids']):
+                        if int(cogid):
+                            ds.add_cognate(
+                                lexeme=lex,
+                                Cognateset_ID=cogid,
+                                Segment_Slice=morpheme_index + 1,
+                                Alignment=data[idx, 'alignment'].split(' + ')[morpheme_index].split(),
```

Comment:
Here's one problem with the alignment and our cldf specification. I used the mapper above, in order to de-segment units that Tim had merged together, like "pl", although they are no clusters. They just make sense for the purpose of comparison, but BIPA would not accept them. So the mapper must, I am afraid, also be applied to "alignments" (and we'll loose the "structure", but we don't really need it now anyway).

Reply:
Ah ok. Will apply these mappers as well. Shouldn't they be implemented as orthography profile?

Reply:
Sorry, forgot to reply immediately. In some sense, yes, we could add them as orthoprofile, we actually have one, but then I'd have to go back to the original forms, and Tim has changed them in the meantime, so the original orthoprofile served its purpose, but now we have the expert segmentation, manually done, with the problem of the changes as indicated in the mapper.

Reply:
It's a tricky situation where we profit from people working in our frameworks, but evolving them in some sense: for alignments, counting "pl" as one segment makes absolute sense in this language group, but for comparison across languages, it does not. We could reverse-back-engineer an orthography profile now, but this would also be a bit strange, right?

Reply:
I'll just add this change, and see how it goes. I want to use your new constructs for @tresoldi's yanglalo data as well, as there we have problems with splitting forms...